nixos/modules/services/networking/hostapd.nix

```diff
@@ -1025,8 +1025,7 @@ in {
                     # Add sae passwords from file
                     "20-saePasswordsFile" = mkIf (bssCfg.authentication.saePasswordsFile != null) (pkgs.writeShellScript "sae-passwords-file" ''
                       HOSTAPD_CONFIG_FILE=$1
-                      grep -v '\s*#' ${escapeShellArg bssCfg.authentication.saePasswordsFile} \
-                        | sed 's/^/sae_password=/' >> "$HOSTAPD_CONFIG_FILE"
+                      echo sae_password_file=${escapeShellArg bssCfg.authentication.saePasswordsFile} >> "$HOSTAPD_CONFIG_FILE"
```

Comment:
Using `sae_password_file` will break existing setups that use comments in their password file. The logic was added to explicitly allow comment but you are right in that the regex is currently wrong.

A better solution IMO would be to just use the correct regex to `^\s*#`.

Reply:
While that does fix the issue to allow passwords with the octothorpe character to not be ignored, my understanding is that comments can be included at the end of lines as well (basing this off of [this documentation](https://man.openbsd.org/hostapd.conf.5), LMK if there's somewhere better to look). In addition, I'm just not sure it feels so good to try re-implement the config parsing logic with regex, as there appears to be some more complex rules regarding multi-line values that wouldn't be handled by this regex. For the time being, I've changed my config to use `services.hostapd.radios.<name>.networks.<name>.settings.sae_password_file = /path/to/file`.

Reply:
I checked the source code yesterday and found no part where they handle comments in the sae password file. If you want to double check [here is the function](https://w1.fi/cgit/hostap/tree/hostapd/config_file.c?h=hostap_2_11#n2340) that parses the sae password file.

> as there appears to be some more complex rules regarding multi-line values that wouldn't be handled by this regex.

Hostapd's config parser does everything on a strict by-line basis, what type of multi-line value are you referring to? I guess you can break up a line using `\` at the end, but that only applies to the main file.

> In addition, I'm just not sure it feels so good to try re-implement the config parsing logic with regex

In principle I agree, however in the specific case of hostapd I don't think we have a choice. Their proprietary configuration file format leaves no room to reuse a parser since everything is custom and non-standard. So regex is probably the best thing we have. I really encourage you to skim over the file I linked and you will understand what I mean - their config parser is essentially one big global state object manipulated by a list of ~550 chained `if .. else if` blocks (further down in that file). There are just no guarantee for anything really, but I guess they don't ever want to refactor so it's likely set in stone forever.

If you know of a more robust way to handle this than regex, I'm of course open to suggestions.

Reply:
> what type of multi-line value are you referring to?

The link to hostapd documentation I posted (which very well may be incorrect) mentioned the possibility of multi-line values by using a backslash at the end of a line.

Thanks for tracking down the parser for that particular `sae_password_file` config, quite unfortunate that they have a special parser just for that file and it doesn't seem to share the same syntax as the rest of their config. I definitely trust your judgement on this, you definitely have lots of familiarity with the ways of hostapd. I think my main confusion in this was that the nixos option `saePasswordsFile` had a similar name to the `sae_password_file` config supported directly by hostapd, yet different syntax.

Reply:
I've updated the PR to go with the recommended regex, as this better fits the nixos option's documentation.

Reply:
> I think my main confusion in this was that the nixos option saePasswordsFile had a similar name to the sae_password_file config supported directly by hostapd, yet different syntax.

Yeah that's an unfortunate clash with the nixos convention to name stuff `xyzFile` if it reads `xyz` from a file at runtime.